### PR TITLE
feat(layout): Easily identify a flexbox area on page so that can start…

### LINKED ISSUE #37: Easily identify a flexbox area on page so that can start adding components
## User Story

As a site builder/editor, I want to easily see that I can click on flexbox area and start adding components.

Wireframe: https://jhdfdc.axshare.com/#g=1&p=flexbox&id=bqple5 

 

### Acceptance Criteria

AC1: IF the user is on a page that has an empty Flexbox
THEN only on the empty Flexbox(s) there is visual representation indicating what area can be worked via the Flexbox

AC2: The design of the empty flexbox area matches what is in the wireframes (see attached screenshot)

Has a dotted line around the area
Has a placeholder message within the box area "Empty Flexbox"
The placeholder is translatable 

AC3: When the user clicks in the outlined flexbox area then the + is added to the menu bar and works as before

AC4: This change does not render anything on the static site if nothing has been added via the flexbox. 

AC5. The component used to render the placeholder is injectable via the ui prop.

### PR DESCRIPTION
**Pull request by dewen**
… adding components.

## Changes

This feature adds visual representation indicating what area can be worked via the Flexbox. When the user clicks in the outlined flexbox area then the + is added to the menu bar and works as before

### Test 1.
- Check if no items inside Flexbox container, there are design updates applied.

### Test 2.
- The behavior is kept the same if there are items added.

### Test 3.
- Static site does not render the design updates nor placeholder.

## Related Issues

Fixes #37

### CODE REVIEW

File: packages/bodiless-layouts/src/FlexboxGrid/SortableContainer.tsx [hunk 1]
```diff
@@ -21,9 +21,14 @@ type SortableListProps = {
 };
 
 const SortableListWrapper = SortableContainer(
-  ({ children }: SortableListProps): React.ReactElement<SortableListProps> => (
-    <section className="bl-flex bl-flex-wrap bl-py-grid-3">{children}</section>
-  ),
+  ({ children }: SortableListProps): React.ReactElement<SortableListProps> => {
+    let className = 'bl-flex bl-flex-wrap bl-py-grid-3';
+    if (!children || !children.length) {
+      className = `${className} bl-justify-center bl-border-2 bl-border-dashed text-gray-600`;
```
Comment (hvanyo) [inline packages/bodiless-layouts/src/FlexboxGrid/SortableContainer.tsx]: Any classes adding in bodiless should be prefaced with bl.

File: packages/bodiless-layouts/src/FlexboxGrid/SortableContainer.tsx [hunk 2]
```diff
@@ -21,9 +21,14 @@ type SortableListProps = {
...
+      return (<section className={className}>Empty</section>);
+    }
+    return (<section className={className}>{children}</section>);
```
Comment (wodenx) [inline packages/bodiless-layouts/src/FlexboxGrid/SortableContainer.tsx]: This is the component that needs to be injected via ui prop.  You'll need to pass the ui which [EditFlexbox receives as a prop](https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-layouts/src/FlexboxGrid/EditFlexbox.tsx#L49) to SortableContainer [here](https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-layouts/src/FlexboxGrid/EditFlexbox.tsx#L64); and you'll need to add this component to the [UI type](https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-layouts/src/FlexboxGrid/types.ts#L21) and provide a default, as is done [for SortableResizableWrapper](https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-layouts/src/SortableResizableWrapper/index.tsx#L41-L55).  You'll then need to add the styled version to the UI which is passed in by [@bodiless/layouts-ui](https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-layouts-ui/src/FlexboxGrid.tsx#L20-L23)

**Comment by VancheeZze** [comment]
The PR changes look good, but the build is failing.

**Comment by dewen** [comment]
> Hi @dewen,
> 
> one minor issue is found:
> when the empty flexbox area is selected/in focus, it doesn't match the designs.
> Please see screenshot:
> Screen Shot 2020-02-17 at 6 19 45 PM
> 
> Everything else looks and works as expected.
> 
> Thanks,
> Tatsiana

Updated

**Pull request merged**